src/context-parser.js

```diff
@@ -611,55 +616,52 @@ Parser.prototype.getCurrentState = function() {
 /**
  * @function Parser#getStates
  *
- * @returns {Array} An array of states.
+ * @returns {Array} An array of states, or undefined if enableStateTracking is off
  *
  * @description
  * Get the states of the HTML5 page
  *
  */
 Parser.prototype.getStates = function() {
-    return this.states.slice();
+    return this.states && this.states.slice();
 };
 
 /**
  * @function Parser#setInitState
+ * @alias Parser#reset
  *
  * @param {integer} state - The initial state of the HTML5 Context Parser.
  *
  * @description
- * Set the init state of HTML5 Context Parser.
+ * Set the init state of HTML5 Context Parser. Once this is called, any existing states will be reset, in order to make use of the init state for processing.
  *
  */
-Parser.prototype.setInitState = function(state) {
-    this.states = [state];
-    return this;
-};
+Parser.prototype.setInitState = Parser.prototype.reset;
```

Comment:
`.setInitState()` is a bit special. the proposed change here is to migrate `.setInitState()` with `.reset()`. note `.reset()` now takes initState.

Rationale: developers would like to set a new init state, and likely he'll then `contextualize()` with such a newly init state.

If `contextualize()` is never called before:
- with state tracking, we should at least change: `this.state=initState; this.states = [initState]`
- without state tracking, we should change `this.state=initState`. This is actually equiv. to `setCurrentState()`

What if `contextualize()` is called once already, developers then call `setInitState()` and then `contextualize()`.
In this regard, reseting everything (like tag and attr name caches) using `reset()` makes sense to ensure that the newly configured initState is independent of any previous calls.